handlers/handlers.go

```diff
@@ -134,12 +134,16 @@ func (r RemoveHandler) ShouldHandle() bool {
 
 func (r RemoveHandler) Handle(m *monitors.Monitors) bool {
 	for _, monitor := range *m {
+        if len(monitor.EmptyDesktops()) == 1 {
+            return true
+        }
 		for _, desktop := range monitor.EmptyDesktops() {
-			if *desktop == monitor.Desktops[len(monitor.Desktops)-1] {
+			if r.config.Min >= len(monitor.Desktops) {
 				continue
 			}
 
-			if r.config.Min >= len(monitor.Desktops) {
+			// TODO: Should we handle desktop destruction if the monitor focus is switched?
+			if !r.config.RemoveFocused && monitor.FocusedDesktopId == desktop.Id {
```

Comment:
@cmschuetz One side effect of this is that an empty desktop could end up in the middle of the desktops.
I.E. where bold means not empty
"**1** **2** **3** 4"
// close window in 2, 2 is focused so `btops` removes 4
"**1** 2 **3**"

I suggest adding in a `MoveHandler`, and having a config option `move-empty` where a value such as `first, last, keep` would move the empty desktop to the beginning, end, or not move the desktop (respectively)
A `MoveHandler` may eventually be used along with the `RenameHandler` to move a desktop to the specified position if it contains a window or has a name, i.e:
```
[[names.classified]]
"web" = ["Chromium", "Firefox"]
position = 0


[[names.classified]]
"" = ["Termite", "Alacritty", "URxvt"]
position = 1
```
This would share a similar `priority` metric as `RenameHandler` would (if I'm skimming the code correctly) in addition to a check if the position is available (i.e. if there's only 1 desktop, no movement operations are needed, or if  the position requested is 5, but there's only 3 desktops, move the desktop to the end)
One unexpected side effect from the second example would be the case where there are 4 desktops, a new one is created, and the empty desktop is switched with the position 5 desktop. Another option could be added to swap with empty desktops...

Reply:
I think we accomplish the desired outcome with this line as-is without the need for adding a new handler.  Essentially, we want this function to take no action if `remove-focused == false && currently focused desktop is empty`.  Once we focus a new desktop, bspwm will send us another message and your `TODO` comment is handled automatically.  I was able to reach what I believe is your desired outcome with:

```
func (r RemoveHandler) Handle(m *monitors.Monitors) bool {
	for _, monitor := range *m {
		for _, desktop := range monitor.EmptyDesktops() {
			if *desktop == monitor.Desktops[len(monitor.Desktops)-1] {
				continue
			}

			if !r.config.RemoveFocused && monitor.FocusedDesktopId == desktop.Id {
				continue
			}

			if r.config.Min >= len(monitor.Desktops) {
				continue
			}

			err := monitor.RemoveDesktop(desktop.Id)
			if err != nil {
				log.Println("Unable to remove desktop: ", desktop.Name, err)
				continue
			}

			return true
		}
	}

	return false
}
```

With the above, if desktop `2` is empty, neither `2` nor `4` are removed.  Once I switch to another desktop, `2` is cleaned up as desired.  Is there another case i'm not considering here?

Reply:
This seems like it would do it.
I wanted to wait on some feedback on the expected behavior on empty desktop removal priority (or expectation?) before continuing with more logic.